terraform/modules/hetzner-kubernetes/load-balancer.resources.tf

```diff
@@ -0,0 +1,80 @@
+resource "hcloud_load_balancer" "lb" {
+  count = var.with_load_balancer ? 1 : 0
+
+  name     = "${var.cluster_name}-lb"
+  location = var.default_location
+
+  load_balancer_type = (
+    length(var.machines) <= 5 ? "lb11" : (
+    length(var.machines) <= 75 ? "lb21" : (
+    length(var.machines) <= 150 ? "lb31" : null))
+  )
+
+  # NOTE: not sure what impact the fact has that one LB targets two possibly
+  #       disjunct groups of machines and what role the algo play in that
+  algorithm {
+    type = "round_robin"
+  }
+}
+
+
+resource "hcloud_load_balancer_network" "lb-nw" {
```

Comment:
I think this should preferably not live in the kubernetes module, but in a separate `private_network` module and then be an input to the kubernetes module.

Say I have another module defining `cassandra` cluster or `elasticsearch` cluster; I want these to be in the same private network as the kubernetes cluster.  

Another option would be to let the `kubernetes` module provide the private network as an output, but this is harder for refactoring and a bit awkward (Why would the `cassandra` module have to depend on the `kubernetes` module; for example)

See Also See the section about [Dependency Inversion](https://www.terraform.io/docs/language/modules/develop/composition.html#dependency-inversion) in the modules best practises guide.

Reply:
Scrap this comment I put it on the wrong part of the file